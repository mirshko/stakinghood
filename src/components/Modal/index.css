[data-reach-dialog-content] {
  padding: 0;
  max-width: 480px;
  width: 100%;
}

@media only screen and (max-width: 40rem) {
  [data-reach-dialog-content] {
  }
}
