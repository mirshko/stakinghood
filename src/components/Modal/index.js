import React from "react";
import { Dialog as ReachDialog } from "@reach/dialog";
import styled from "styled-components";
import PropTypes from "prop-types";
import { Flex } from "rebass";

import UnstyledButton from "../UnstyledButton";

import "@reach/dialog/styles.css";
import "./index.css";
import Uppercase from "../Uppercase";

const Dialog = styled(ReachDialog)`
  border-radius: 8px;
  box-shadow: rgba(94, 111, 243, 0.16) 0px 48px 80px,
    rgba(22, 25, 49, 0.16) 0px 16px 32px, rgba(22, 25, 49, 0.08) 0px 6px 16px;
`;

const Modal = ({ children, onClose }) => (
  <Dialog>
    {children}
    <Flex justifyContent="center" alignItems="center" pb={3}>
      <UnstyledButton width="initial" onClick={onClose}>
        <Uppercase color="var(--brand)">Close</Uppercase>
      </UnstyledButton>
    </Flex>
  </Dialog>
);

Modal.propTypes = {
  children: PropTypes.node.isRequired,
  onClose: PropTypes.func.isRequired
};

export default Modal;
