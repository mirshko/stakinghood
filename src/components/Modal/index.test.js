import React from "react";
import { shallow } from "enzyme";

import Modal from "./index";

describe("<Modal />", () => {
  it("it works", () => {
    const wrapper = shallow(
      <Modal onClick={() => console.log("Close Modal")}>
        <p>Children</p>
      </Modal>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
