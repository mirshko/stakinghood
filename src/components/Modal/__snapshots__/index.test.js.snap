// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> it works 1`] = `
<Dialog>
  <p>
    Children
  </p>
  <Styled(styled.div)
    alignItems="center"
    justifyContent="center"
    pb={3}
  >
    <UnstyledButton
      onClick={[Function]}
      width="initial"
    >
      <Styled(Styled(styled.div))
        color="var(--brand)"
      >
        Close
      </Styled(Styled(styled.div))>
    </UnstyledButton>
  </Styled(styled.div)>
</Dialog>
`;
